internal/service/base/resource_language_translation.go

```diff
@@ -0,0 +1,507 @@
+	if r.Client == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialised",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.",
+		)
+		return
+	}
+}
+
+type languageTranslationResourceModel struct {
+	EnvironmentId  pingonetypes.ResourceIDValue `tfsdk:"environment_id"`
+	Id             pingonetypes.ResourceIDValue `tfsdk:"id"`
+	Key            types.String                 `tfsdk:"key"`
+	Locale         types.String                 `tfsdk:"locale"`
+	ReferenceText  types.String                 `tfsdk:"reference_text"`
+	ShortKey       types.String                 `tfsdk:"short_key"`
+	TranslatedText types.String                 `tfsdk:"translated_text"`
+}
+
+func (r *languageTranslationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+
+	resp.Schema = schema.Schema{
+		Description: "Resource to create and manage the language translation.",
+		Attributes: map[string]schema.Attribute{
+			"environment_id": framework.Attr_LinkID(
+				framework.SchemaAttributeDescriptionFromMarkdown("The ID of the environment to create and manage the language_translation in."),
+			),
+			"id": framework.Attr_ID(),
+			"key": schema.StringAttribute{
+				Required:    true,
+				Description: "The page and name of the interface element to be localized (for example, `flow-ui.button.cancel`).",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"locale": schema.StringAttribute{
+				Required:    true,
+				Description: fmt.Sprintf("An ISO standard language code. For more information about standard language codes, see [ISO Language Code Table](http://www.lingoes.net/en/translator/langcode.htm).  The following language codes are reserved as they are created automatically in the environment: %s.", verify.IsoReservedListString()),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						verify.ReservedIsoList()...,
+					),
+				},
+			},
+			"reference_text": schema.StringAttribute{
+				Computed:    true,
+				Description: "The English string text associated with the interface element.",
+			},
+			"short_key": schema.StringAttribute{
+				Computed:    true,
+				Description: "The interface element only (for example, `button.cancel`).",
+			},
+			"translated_text": schema.StringAttribute{
+				Required:    true,
+				Description: "The translated string text associated with the interface element.",
+			},
+		},
+	}
+}
+
+func (model *languageTranslationResourceModel) buildClientStruct() (*[]management.LocaleTranslation, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	localeTranslationSlice := &[]management.LocaleTranslation{}
+	innerLocaleTranslation := &management.LocaleTranslation{}
+
+	// Validate and set the Key field
+	if model.Key.IsNull() || model.Key.IsUnknown() {
```

Comment:
I don't think we need null/unknown checks for required fields, as far as I know we don't do this for other resources